arch/xtensa/src/esp32s2/esp32s2_spi_slave.c

```diff
@@ -0,0 +1,1734 @@
+
+static void spislave_bind(struct spi_slave_ctrlr_s *ctrlr,
+                          struct spi_slave_dev_s *dev,
+                          enum spi_slave_mode_e mode,
+                          int nbits);
+static void spislave_unbind(struct spi_slave_ctrlr_s *ctrlr);
+static int spislave_enqueue(struct spi_slave_ctrlr_s *ctrlr,
+                            const void *data,
+                            size_t nwords);
+static bool spislave_qfull(struct spi_slave_ctrlr_s *ctrlr);
+static void spislave_qflush(struct spi_slave_ctrlr_s *ctrlr);
+static size_t spislave_qpoll(struct spi_slave_ctrlr_s *ctrlr);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/*  SPI2 private data */
+
+#ifdef CONFIG_ESP32S2_SPI2
+static const struct spislave_config_s esp32s2_spi2slave_config =
+{
+  .width        = SPI_SLAVE_DEFAULT_WIDTH,
+  .mode         = SPI_SLAVE_DEFAULT_MODE,
+  .id           = 2,
+  .cs_pin       = CONFIG_ESP32S2_SPI2_CSPIN,
+  .mosi_pin     = CONFIG_ESP32S2_SPI2_MOSIPIN,
+  .miso_pin     = CONFIG_ESP32S2_SPI2_MISOPIN,
+  .clk_pin      = CONFIG_ESP32S2_SPI2_CLKPIN,
+  .periph       = ESP32S2_PERIPH_SPI2,
+  .irq          = ESP32S2_IRQ_SPI2,
+  .clk_bit      = SYSTEM_SPI2_CLK_EN,
+  .rst_bit      = SYSTEM_SPI2_RST,
+#ifdef CONFIG_ESP32S2_SPI_DMA
+  .dma_clk_bit  = SYSTEM_SPI2_DMA_CLK_EN,
+  .dma_rst_bit  = SYSTEM_SPI2_DMA_RST,
+  .dma_periph   = ESP32S2_DMA_PERIPH_SPI2,
+#endif
+  .cs_insig     = FSPICS0_IN_IDX,
+  .cs_outsig    = FSPICS0_OUT_IDX,
+  .mosi_insig   = FSPID_IN_IDX,
+  .mosi_outsig  = FSPID_OUT_IDX,
+  .miso_insig   = FSPIQ_IN_IDX,
+  .miso_outsig  = FSPIQ_OUT_IDX,
+  .clk_insig    = FSPICLK_IN_IDX,
+  .clk_outsig   = FSPICLK_OUT_IDX,
+};
+
+static const struct spi_slave_ctrlrops_s esp32s2_spi2slave_ops =
+{
+  .bind     = spislave_bind,
+  .unbind   = spislave_unbind,
+  .enqueue  = spislave_enqueue,
+  .qfull    = spislave_qfull,
+  .qflush   = spislave_qflush,
+  .qpoll    = spislave_qpoll
+};
+
+#ifdef CONFIG_ESP32S2_SPI_DMA
+
+/* SPI DMA RX/TX description buffer */
+
+static struct esp32s2_dmadesc_s esp32s2_spi2_dma_rxdesc[SPI_DMA_DESC_NUM];
+static struct esp32s2_dmadesc_s esp32s2_spi2_dma_txdesc[SPI_DMA_DESC_NUM];
+#endif
+
+static struct spislave_priv_s esp32s2_spi2slave_priv =
+{
+  .ctrlr         =
+                  {
+                    .ops = &esp32s2_spi2slave_ops
+                  },
+  .dev           = NULL,
+  .config        = &esp32s2_spi2slave_config,
+  .refs          = 0,
+  .cpu           = -1,
+  .cpuint        = -ENOMEM,
+#ifdef CONFIG_ESP32S2_SPI_DMA
+  .dma_channel   = -ENOMEM,
+  .dma_rxdesc    = esp32s2_spi2_dma_rxdesc,
+  .dma_txdesc    = esp32s2_spi2_dma_txdesc,
+  .rx_dma_offset = 0,
+#endif
+  .mode          = SPISLAVE_MODE0,
+  .nbits         = 0,
+  .tx_length     = 0,
+  .tx_buffer     =
+                  {
+                    0
+                  },
```

Comment:
actually zero init to fields is not needed as those will be zero inialized at boot